From 499ec4e50734039504af732964985a3e094434ee Mon Sep 17 00:00:00 2001
From: Simon Richter <Simon.Richter@hogyros.de>
Date: Mon, 13 May 2013 23:02:16 +0200
Subject: [PATCH 04/15] Make libgbm optional without EGL support
To: beignet@lists.freedesktop.org

If EGL or GBM cannot be found, the EGL support is disabled, and then
neither library is required.
---
 src/CMakeLists.txt |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

Index: beignet-0.1+git20130514+19e9c58/src/CMakeLists.txt
===================================================================
--- beignet-0.1+git20130514+19e9c58.orig/src/CMakeLists.txt	2013-05-14 20:05:52.000000000 +0200
+++ beignet-0.1+git20130514+19e9c58/src/CMakeLists.txt	2013-05-14 20:08:57.006022846 +0200
@@ -28,14 +28,16 @@
     x11/dricommon.c 
     x11/va_dri2.c)
 
-if (EGL_FOUND)
+if (EGL_FOUND AND GBM_FOUND)
 set (OPENCL_SRC ${OPENCL_SRC} cl_mem_gl.c cl_gl_api.c x11/gbm_dri2_x11_platform.c)
 SET(CMAKE_CXX_FLAGS "-DHAS_EGL ${CMAKE_CXX_FLAGS}")
 SET(CMAKE_C_FLAGS "-DHAS_EGL ${CMAKE_C_FLAGS}")
 SET(OPTIONAL_EGL_LIBRARY "${EGL_LIBRARY}")
-else(EGL_FOUND)
+SET(OPTIONAL_GBM_LIBRARY "${GBM_LIBRARY}")
+else(EGL_FOUND AND GBM_FOUND)
 SET(OPTIONAL_EGL_LIBRARY "")
-endif (EGL_FOUND)
+SET(OPTIONAL_GBM_LIBRARY "")
+endif (EGL_FOUND AND GBM_FOUND)
 
 if (OCLIcd_FOUND)
 set (OPENCL_SRC ${OPENCL_SRC} cl_khr_icd.c)
@@ -57,7 +59,7 @@
                       ${DRM_LIBRARY}
                       ${OPENGL_LIBRARIES}
                       ${OPTIONAL_EGL_LIBRARY}
-                      ${GBM_LIBRARY})
+                      ${OPTIONAL_GBM_LIBRARY})
 set_target_properties(cl
                         PROPERTIES
                         VERSION 0.1
